### PR TITLE
[6.3.x] Bug 1324596 - [GSS] [6.2.z] 6.2.2 unable to open/edit some assets with specific user

### PR DESCRIPTION
**Pull request by Rikkola**
https://bugzilla.redhat.com/show_bug.cgi?id=1324596

Almost the same as master and 6.4.x

Since 6.3.x is frozen. I am not sure if this goes in, but the patch will be done against the 6.3.x. So placing this here for a PR review.

(cherry picked from commit c181908753e20512bad67c342d32c4ae6af13277)

### CODE REVIEW

**Comment by manstis** [comment]
+1 to merge (*if*/*when* the BZ gets blocker+ ack)

**Comment by Rikkola** [comment]
@manstis ok. This has all the flags now and can go in.

File: guvnor-inbox/guvnor-inbox-backend/src/test/java/org/guvnor/inbox/backend/server/security/InboxEntrySecurity_InboxEntrySecurityTestTest.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.java.nio.file.FileSystem;
+import org.uberfire.java.nio.fs.file.SimpleFileSystemProvider;
+import org.uberfire.security.authz.AuthorizationManager;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith( MockitoJUnitRunner.class )
+public class InboxEntrySecurity_InboxEntrySecurityTestTest {
```
Comment (psiroky) [inline guvnor-inbox/guvnor-inbox-backend/src/test/java/org/guvnor/inbox/backend/server/security/InboxEntrySecurity_InboxEntrySecurityTestTest.java]: I've noticed this only now, but the class name looks weird.

Comment (Rikkola) [inline guvnor-inbox/guvnor-inbox-backend/src/test/java/org/guvnor/inbox/backend/server/security/InboxEntrySecurity_InboxEntrySecurityTestTest.java]: It does, let me fix that.

**Comment by Rikkola** [comment]
Fixes done.

File: guvnor-test-utils/guvnor-test-utils.iml [hunk 2]
```diff
@@ -0,0 +1,107 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (manstis) [inline guvnor-test-utils/guvnor-test-utils.iml]: Are you sure!?

File: guvnor-inbox/guvnor-inbox-backend/src/test/java/org/guvnor/inbox/backend/server/security/InboxEntrySecurity_getInboxEntryRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.java.nio.file.FileSystem;
+import org.uberfire.java.nio.fs.file.SimpleFileSystemProvider;
+import org.uberfire.security.authz.AuthorizationManager;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith( MockitoJUnitRunner.class )
+public class InboxEntrySecurity_getInboxEntryRepositoryTest {
```
Comment (manstis) [inline guvnor-inbox/guvnor-inbox-backend/src/test/java/org/guvnor/inbox/backend/server/security/InboxEntrySecurity_getInboxEntryRepositoryTest.java]: Please confirm this class name is intentional - I don't want @psiroky saying it looked funny again ;)

File: guvnor-test-utils/target/checkstyle-checker.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0"?>
```
Comment (manstis) [inline guvnor-test-utils/target/checkstyle-checker.xml]: Likewise.. not sure you want to commit this file?

File: guvnor-test-utils/target/checkstyle-rules.xml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0"?>
```
Comment (manstis) [inline guvnor-test-utils/target/checkstyle-rules.xml]: Likewise.. not sure you want to commit this file?

File: guvnor-test-utils/target/checkstyle.log [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (manstis) [inline guvnor-test-utils/target/checkstyle.log]: Likewise.. not sure you want to commit this file?

**Comment by manstis** [comment]
@Rikkola oeee.. I stopped commenting individually.. looks like you've included a bunch of files in error.

**Comment by Rikkola** [comment]
Need to see what happened here. Looks like another branch is leaking into this one.

File: guvnor-inbox/guvnor-inbox-backend/src/test/java/org/guvnor/inbox/backend/server/security/InboxEntrySecurity_getInboxEntryRepositoryTest.java (hunk 3 above)
Comment (Rikkola) [inline guvnor-inbox/guvnor-inbox-backend/src/test/java/org/guvnor/inbox/backend/server/security/InboxEntrySecurity_getInboxEntryRepositoryTest.java]: It is. I have used the ClassName_methodName a couple of times for tests that only test one method. This makes the method the unit I am testing.

InboxEntrySecurityTest already existed, but the setup in that class is already about 50 lines and the method we are testing does not care about most of the parameters given to it. Also the InboxEntrySecurityTest overwrites this method for testing. So made sense to separate the method to a new test class.

**Comment by Rikkola** [comment]
Looks like the test-utils from guvnor tried to get in. Should be ok now.

**Comment by manstis** [comment]
@mswiderski @mrietveld jBPM has a compilation error on 6.3.x:

```08:52:28 [ERROR] /home/jenkins/workspace/guvnor-pullrequests-6.x/upstream-repos/jbpm/jbpm-flow/src/main/java/org/jbpm/process/StatefulProcessSession.java:[60,7] error: StatefulProcessSession is not abstract and does not override abstract method delete(FactHandle,State) in EntryPoint```

Could you please take a look?

**Comment by Rikkola** [comment]
This build worked locally for me.

**Comment by mswiderski** [comment]
there is a missing commit in jbpm repo after back porting of drools part, @mariofusco just pushed the fix https://github.com/droolsjbpm/jbpm/commit/d516ab507647ad487b879f275bd43e114762ff50

**Comment by mswiderski** [comment]
jenkins retest this

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
@mariofusco Hi Mario, there's still some compilation failures (presumably relating to the recent change you made):

```10:16:07 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.5.1:compile (default-compile) on project kie-remote-client: Compilation failure
10:16:07 [ERROR] /home/jenkins/workspace/guvnor-pullrequests-6.x/upstream-repos/droolsjbpm-integration/kie-remote/kie-remote-client/src/main/java/org/kie/services/client/api/command/KieSessionClientCommandObject.java:[65,7] error: KieSessionClientCommandObject is not abstract and does not override abstract method delete(FactHandle,State) in EntryPoint```

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
Jenkins please retest this.

**Pull request merged**